configure.ac

```
@@ -68,19 +68,22 @@ AC_USE_SYSTEM_EXTENSIONS
 
 AC_LANG([C++])
 
-# Clang 3.2 on some CPUs requires -march-native to detect correctly.
-# GCC 4.3+ can also produce faster executables when its used.
+# Some compilers can produce faster executables when its used.
 # But building inside a virtual machine environment has been found to
 # cause random Illegal Instruction errors due to mis-detection of CPU.
 AC_ARG_ENABLE(arch-native,
-  AS_HELP_STRING([--disable-arch-native],[Some compilers offer CPU-specific
+  AS_HELP_STRING([--enable-arch-native],[Some compilers offer CPU-specific
                  optimizations with the -march=native parameter.
-                 This flag disables the optimization. The default is to
-                 auto-detect compiler support and use where available.]), [
+                 This flag enables the optimization, by default disabled,
+                 provided that the compiler supports it. Enabling this
+                 optimization may cause crashes due to illegal instruction errors
+                 when Squid is run on a system different from the one it is built on,
+                 and in some containerised or virtualised environments]), [
   SQUID_YESNO([$enableval],[--enable-arch-native])
 ])
-AC_MSG_NOTICE([CPU arch native optimization enabled: ${enable_arch_native:=auto}])
-AS_IF([test "x${enable_arch_native}" != "xno"],[
+AC_MSG_NOTICE([CPU arch native optimization enabled: ${enable_arch_native:=no}])
+AS_IF([test "x${enable_arch_native}" = "xyes"],[
+  # XXX: make this a tristate yes/no/auto
```

As previously [requested](https://github.com/squid-cache/squid/pull/2261/files#r2389140818), please add an XXX:

```suggestion
  # XXX: Fail if the explicitly requested native optimization is not available.
  # TODO: Make this a tristate yes/no/auto.
```

The above suggestion also replaces XXX with TODO because the lack of tristate support is not a bug.

P.S. The tristate enhancement is orthogonal to the XXX I am asking for. Even after/if we support `auto`, `yes` should fail if the requested compiler option is not supported by the compiler.